pkg/issuer/acme/dns/dns.go

```diff
@@ -346,23 +344,25 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1.GenericIssuer
 			secretAccessKey = string(secretAccessKeyBytes)
 		}
 
-		webIdentityToken := ""
+		var webIdentityTokenRetriever stscreds.IdentityTokenRetriever
 		if providerConfig.Route53.Auth != nil && providerConfig.Route53.Auth.Kubernetes != nil && providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef != nil {
 			if providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef.Name == "" {
 				return nil, nil, fmt.Errorf("service account name is required for Kubernetes auth")
 			}
 
+			// In case the custom resource was created before the
+			// `+default=["sts.amazonaws.com"]` marker was added to the CRD.
 			audiences := []string{"sts.amazonaws.com"}
 			if len(providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef.TokenAudiences) != 0 {
 				audiences = providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef.TokenAudiences
 			}
 
-			jwt, err := s.createToken(ctx, resourceNamespace, providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef.Name, audiences)
-			if err != nil {
-				return nil, nil, fmt.Errorf("error getting service account token: %w", err)
+			webIdentityTokenRetriever = &route53.KubernetesServiceAccountTokenRetriever{
+				ServiceAccountName: providerConfig.Route53.Auth.Kubernetes.ServiceAccountRef.Name,
+				Audiences:          audiences,
+				Namespace:          resourceNamespace,
+				Client:             s.Client.CoreV1().ServiceAccounts(resourceNamespace),
 			}
```

Comment:
Instead of creating a Kubernetes service account token every time we instantiate the solver,
this instead provides a custom [IdentityTokenRetriever](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/credentials@v1.17.27/stscreds#IdentityTokenRetriever) which the AWS-SDK can call as and when it needs the token.

This also allows the AWS client to refresh the STS token if the DNS-01 prepare or cleanup operations are delayed or if an AWS endpoint is unreachable or returns an error and the AWS client needs to retry any of the STS or Route53 operations.